mmv1/products/edgecontainer/Cluster.yaml

```diff
@@ -0,0 +1,492 @@
+
+--- !ruby/object:Api::Resource
+name: 'Cluster'
+description: |
+  Cluster contains information about a Google Distributed Cloud Edge Kubernetes cluster.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create and manage clusters': 'https://cloud.google.com/distributed-cloud/edge/latest/docs/clusters'
+  api: 'https://cloud.google.com/distributed-cloud/edge/latest/docs/reference/container/rest/v1/projects.locations.clusters'
+base_url: 'projects/{{project}}/locations/{{location}}/clusters'
+self_link: 'projects/{{project}}/locations/{{location}}/clusters/{{name}}'
+create_url: 'projects/{{project}}/locations/{{location}}/clusters?cluster_id={{name}}'
+update_url: 'projects/{{project}}/locations/{{location}}/clusters/{{name}}'
+update_verb: :PATCH
+update_mask: true
+delete_url: 'projects/{{project}}/locations/{{location}}/clusters/{{name}}'
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The location of the resource.
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The GDCE cluster name.
+import_format: ["projects/{{project}}/locations/{{location}}/clusters/{{name}}"]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_cluster"
+    primary_resource_id: "default"
+    vars:
+      edgecontainer_cluster_name: "basic-cluster"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_cluster_with_maintenance_window"
+    primary_resource_id: "default"
+    vars:
+      edgecontainer_cluster_name: "cluster-with-maintenance"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_local_control_plane_cluster"
+    primary_resource_id: "default"
+    skip_test: true
+    vars:
+      edgecontainer_cluster_name: "local-control-plane-cluster"
+properties:
+  - !ruby/object:Api::Type::Time
+    name: "createTime"
+    description: |
+      The time the cluster was created, in RFC3339 text format.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: "updateTime"
+    description: |
+      The time the cluster was last updated, in RFC3339 text format.
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: "labels"
+    default_from_api: true
```

Comment:
Per our conversation, this should probably use a diff suppress to allow server-owned labels as well as user-owned labels. See mmv1/templates/terraform/constants/cloud_run_domain_mapping.go.erb for an example.

Reply:
The backend logic will return the system labels + user defined labels, so I think mark it as default_from_api: true will always take as the source of truth.

Reply:
That will create a problem though, because Terraform will try to overwrite the entire set of labels to only include what is defined by the user. The custom logic would tell Terraform to ignore certain values that are controlled by the server.

Reply:
Thanks for the explanation. Added `diff_suppress_func: 'DomainMappingLabelDiffSuppress'` in cluster/node-pool label fields.

Reply:
Ah ok, makes sense, thanks!